src/layman/uuid.py

```diff
@@ -156,9 +157,17 @@ def check_redis_consistency(expected_publ_num_by_type=None):
     ]
 
     for workspace, publ_type_name, pubname in total_publs:
+        attempt = 1
         chain_info = celery_util.get_publication_chain_info(workspace, publ_type_name, pubname)
         is_ready = celery_util.is_chain_ready(chain_info)
-        assert chain_info['finished'] is is_ready
+        if is_ready and not chain_info['finished']:
```

Comment:
Rewrite this with critical section in is_chain_ready, so we will not get this inconsistent results.